Source/Service/HTTPCallback.swift

```diff
@@ -21,11 +21,22 @@ public protocol HTTPCallback {
     func successfulHandler(response: Data?) -> Void
 }
 
-class CotterCallback: HTTPCallback {
+public protocol InternalCallback {
+    // internalErrorHandler handles any internal error (i.e. keychain error)
+    // prior to HTTP Requests
+    func internalErrorHandler(err: String?) -> Void
+    
+    // internalSuccessHandler handles successful HTTP Response Request
+    func internalSuccessHandler() -> Void
+}
+
+class CotterCallback: HTTPCallback, InternalCallback {
```

Comment:
make an extension instead of specifying it this way. see how `KeyboardViewDelegate`, `AlertServiceDelegate` is being extended to by `PINViewController.swift`

again, this is for code readability